modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/cluster/storage/ClusterStorage.scala

```diff
@@ -66,34 +66,37 @@ object ClusterStorage {
       def getPeer(id: PeerId): F[Option[Peer]] =
         peers(id).get
 
-      def addPeer(peer: Peer): F[Unit] =
-        setPeer(peer.id)(peer.some)
+      def addPeer(newPeer: Peer): F[Boolean] =
+        updatePeer(newPeer.id) {
+          case Some(currPeer) =>
+            if (currPeer.session < newPeer.session) newPeer.some
```

Comment:
May be now inconsistent with changed logic in `validateHandshake` in regards to `<` vs `<=`

Reply:
I will address this comment soon. I don't have a good idea on how to change this yet.

Reply:
Fixed in the recent changes.